azure-templates/build-op-steps.yml

```diff
@@ -0,0 +1,94 @@
+parameters:
+  - name: projectLocation
+    type: string
+  - name: buildOperation
+    type: string
+  - name: target
+    type: string
+    default: ''
+  - name: buildSpec
+    type: string
+    default: ''
+  - name: dependencies
+    type: object
+    default:
+      - file: ''
+
+steps:
+
+###########  Pull Dependencies in from latest build locations  ###########
+  - ${{ each dependency in parameters.dependencies }}: # TEMPORARY - default branchName is 'azure-pipelines' until things are actually committed to main on dependencies
+    - ${{ if ne( dependency.file, '' )}}:
+      - ${{ if eq( parameters.target, 'My Computer' )}}:
+        - powershell: Write-Output "##vso[task.setvariable variable=dependencyTargetDirectory]Windows"
+          displayName: set target for dependency to Windows
+      - ${{ if eq( parameters.target, 'Linux x64' )}}:
+        - powershell: Write-Output "##vso[task.setvariable variable=dependencyTargetDirectory]Linux_x64"
+          displayName: set target for dependency to Linux RT
+      - powershell: |
+          $branchName = 'azure-pipeline'
+          If (Test-Path -Path '${{ dependency.source }}\NI\export\$(Build.SourceBranchName)*\$(lvVersion)\$(architecture)') 
+          {
+            $branchName = '$(Build.SourceBranchName)'
+          }
+          $dependencyFilePath = Get-ChildItem `
+            -Path "${{ dependency.source }}\NI\export\$branchName\*\$(lvVersion)\$(architecture)\$(dependencyTargetDirectory)\*" `
+            -Recurse `
+            -Include '*${{ dependency.file }}' `
+            | Sort {$_.LastWriteTime} `
```

Comment:
I recall the build tools originally sorting by write time - maybe they still do. It'd be nice if we could find a way to interpret the actual build number and consume the latest... although it isn't critical.

Reply:
I can switch to using the "BuildNumber" which is essentially a date/time, instead of the BuildID which is a unique number for each build that doesn't appear to communicate newness, just uniqueness in the org.  Then we should be able to use the BuildNumber instead of Write time

Reply:
switched to use build number for search and directory in commit 5